### PR TITLE
Google component filters

### LINKED ISSUE #426: Google
In my project I have extended the GoogleMapsProvider to have a setter for component filters.

https://developers.google.com/maps/documentation/geocoding/#ComponentFiltering

If I'd create a PR, would you such kind of additional functionality?

### PR DESCRIPTION
**Pull request by hannesvdvreken**
### Added

* Ability to add component filter to google location search provider. [documentation](https://developers.google.com/maps/documentation/geocoding/#ComponentFiltering)

--
Closes #426

### CODE REVIEW

**Comment by willdurand** [comment]
@geocoder-php/geocoder WDYT?

**Comment by Baachi** [comment]
Sure why not.

File: src/Geocoder/Provider/GoogleMaps.php [hunk 1]
```diff
@@ -261,4 +292,18 @@ private function updateAddressComponent(&$resultSet, $type, $values)
 
         return $resultSet;
     }
+
+    /**
+     * @return string
+     */
+    private function getComponentFilterString()
+    {
+        $parts = [];
```
Comment (Baachi) [inline src/Geocoder/Provider/GoogleMaps.php]: No need for this method, you can simply write:

```php
implode('|', array_map($this->componentFilters, function($value, $key) { 
    return $key.':'.$value; 
}));
```

Comment (hannesvdvreken) [inline src/Geocoder/Provider/GoogleMaps.php]: Also gonna need to urlencode the value.

**Comment by toin0u** [comment]
I agree :+1:

**Comment by hannesvdvreken** [comment]
[Build](https://travis-ci.org/geocoder-php/Geocoder/builds/61027627) failed because `hhvm-nightly` is no longer available in the build environment: https://github.com/travis-ci/travis-ci/issues/3788.

Should be solved by #429.

**Comment by hannesvdvreken** [comment]
Rebased

**Comment by willdurand** [comment]
Could you add some tests please?

**Comment by hannesvdvreken** [comment]
I can, but won't have time for it soon.

**Comment by willdurand** [comment]
@hannesvdvreken this PR still needs tests :)

File: src/Geocoder/Provider/GoogleMaps.php [hunk 2]
```diff
@@ -126,11 +151,20 @@ protected function buildQuery($query)
             $query = sprintf('%s&key=%s', $query, $this->apiKey);
         }
 
+        if (!empty($this->componentFilters)) {
+            $componentFilters = $this->componentFilters;
+            $query = sprintf('%s&key=%s', $query, implode('|', array_map(function($key) use ($componentFilters) {
```
Comment (mvcaaa) [inline src/Geocoder/Provider/GoogleMaps.php]: Must be 
```php
$query = sprintf('%s&components=%s', $query, implode('|', array_map(function($key) use ($componentFilters) {
```

**Comment by willdurand** [comment]
Replaced by #466

**Pull request closed without merging**